### PR TITLE
Issue 10398 delete textarea fields mssql

### PR DESCRIPTION
**Pull request by joseorsini**
Tested on backported fix to 3.6.1, running on SQL Server DB. 
Works OK

### CODE REVIEW

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java [hunk 1]
```diff
@@ -2322,9 +2322,20 @@ private Queries getQueries(Field field) {
...
+            if (!DbConnectionFactory.isMsSql()){
+                whereField.append(field.getFieldContentlet()).append(" != ");
+            } else {
+                //Specific logic for text(varchar) and textarea (text) fields
+                if (field.getFieldContentlet().contains("text_area")){
+                    whereField.append(" DATALENGTH (").append(field.getFieldContentlet()).append(")");
+                } else{
+                    whereField.append(field.getFieldContentlet()).append(" != ");
+                }
+            }
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: I think this if else if else logic can be simplified a bit:
if(DbConnectionFactory.isMsSql() && field.getFieldContentlet().contains("text_area")) {
    whereField.append(" DATALENGTH (").append(field.getFieldContentlet()).append(")");
} else {
  whereField.append(field.getFieldContentlet()).append(" != ");
}

**Comment by joseorsini** [comment]
@dsilvam please review. Suggestions on code were applied. Thanks for the tip!

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java [hunk 2]
```diff
@@ -2322,9 +2322,15 @@ private Queries getQueries(Field field) {
...
         } else {
-            whereField.append(field.getFieldContentlet()).append(" IS NOT NULL AND ").append(field.getFieldContentlet())
-                    .append(" != ");
+            whereField.append(field.getFieldContentlet()).append(" IS NOT NULL AND ");
+            if(DbConnectionFactory.isMsSql() && field.getFieldContentlet().contains("text_area")) {
+                whereField.append(" DATALENGTH (").append(field.getFieldContentlet()).append(")");
+            } else {
+                whereField.append(field.getFieldContentlet()).append(" != ");
+            }
         }
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: Looks good @joseorsini

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java [hunk 3]
```diff
@@ -2345,8 +2356,19 @@ private Queries getQueries(Field field) {
...
+            } else {
+                //Specific logic for text(varchar) and textarea (text) fields
+                if (field.getFieldContentlet().contains("text_area")){
+                    update.append("''");
+                    whereField.append(" > 0");
+                } else{
+                    update.append("''");
+                    whereField.append("''");
+                }
+            }
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: @joseorsini what about here? I think the same thing can be done

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java (hunk 3 above)
Comment (joseorsini) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: @dsilvam please review. PR has been updated.

**Pull request merged**